clang/lib/CIR/Dialect/Transforms/TargetLowering/LowerFunction.cpp

```diff
@@ -193,7 +448,21 @@ LogicalResult LowerFunction::buildFunctionEpilog(const LowerFunctionInfo &FI) {
         return success();
       }
     } else {
-      llvm_unreachable("NYI");
+      // NOTE(cir): Unlike the original codegen, CIR may have multiple return
+      // statements in the function body. We have to handle this here.
+      mlir::PatternRewriter::InsertionGuard guard(rewriter);
+      NewFn->walk([&](ReturnOp returnOp) {
+        rewriter.setInsertionPoint(returnOp);
+
+        // TODO(cir): I'm not sure if we need this offset here or in CIRGen.
```

Comment:
Same here, comments tend to live longer than then activity of people writing them.